lib/ansible/modules/messaging/rabbitmq_user.py

```diff
@@ -229,6 +229,14 @@ def has_tags_modifications(self):
     def has_permissions_modifications(self):
         return sorted(self._permissions) != sorted(self.permissions)
 
+    def has_password_modification(self):
+        _, rc = self._exec(['authenticate_user', self.username, self.password], run_in_check_mode=True, check_rc=False)
+        # 'EX_NOUSER' (The user specified does not exist) from https://github.com/rabbitmq/rabbitmq-server/blob/master/include/rabbit_cli.hrl#L70
+        return rc == 67
```

Comment:
`rabbitmqctl authenticate_user` returns `67` ([EX_NOUSER](https://github.com/rabbitmq/rabbitmq-server/blob/master/include/rabbit_cli.hrl#L70)) when can't authenticate.
Old `rabbitmqctl` (RabbitMQ < 3.6.5) will return `64` ([EX_USAGE](https://github.com/rabbitmq/rabbitmq-server/blob/master/include/rabbit_cli.hrl#L68) exit code) for `authenticate_user` option.